Okta/OktaAuth.swift

```diff
@@ -95,6 +97,57 @@ public struct OktaAuthorization {
             }
         })
     }
+    
+    func authorize(withSessionToken sessionToken: String, config: [String: String]) -> Promise<OktaTokenManager> {
+        return Promise<OktaTokenManager>(in: .background, { resolve, reject, _ in
+            // Discover Endpoints
+            guard let issuer = config["issuer"],
+                  let clientId = config["clientId"],
+                  let redirectUriString = config["redirectUri"],
+                  let redirectUri = URL(string: redirectUriString) else {
+                    return reject(OktaError.missingConfigurationValues)
+            }
+
+            self.getMetadataConfig(URL(string: issuer))
+            .then { oidConfig in
+                let codeVerifier = OIDAuthorizationRequest.generateCodeVerifier()!
+                let codeChallenge = OIDAuthorizationRequest.codeChallengeS256(forVerifier: codeVerifier)
+                let state = OIDAuthorizationRequest.generateState()
+
+                let request = OIDAuthorizationRequest(
+                    configuration: oidConfig,
+                    clientId: clientId,
+                    clientSecret: nil,
+                    scope: config["scopes"],
+                    redirectURL: redirectUri,
+                    responseType: OIDResponseTypeCode,
+                    state: state,
+                    nonce: nil,
+                    codeVerifier: codeVerifier,
+                    codeChallenge: codeChallenge,
+                    codeChallengeMethod: OIDOAuthorizationRequestCodeChallengeMethodS256,
+                    additionalParameters: ["sessionToken" : sessionToken]
+                )
+                
+                OIDAuthState.getState(withAuthRequest: request)
```

Comment:
Just to confirm, this does not open a browser, correct?

Reply:
Yes, correct.